README.md

```diff
@@ -179,3 +180,49 @@ jobs:
           manifest_file: example-app/spin.toml
           fermyon_token: ${{ secrets.FERMYON_CLOUD_TOKEN }}
 ```
+
+
+## `fermyon/action/spin/preview`
+
+Build and deploy the `spin` app preview to Fermyon Cloud
+
+### Inputs
+
+| Name          | Required | Description                                                                       | Default   |
+| ------------- | -------- | --------------------------------------------------------------------------------- | --------- |
+| fermyon_token | Required | Fermyon Cloud Personal access token for deploying the `spin` app to Fermyon Cloud | -         |
+| manifest_file | Optional | Path to `spin.toml`. Used with the `build`/`deploy` command.                      | spin.toml |
+| github_token  | Required | The `GitHub` token for updating comment on PR                                     | -         |
+| undeploy      | Optional | If true, removes the preview deployment from Fermyon Cloud                        | -         |
+
+### Example
+
+```yaml
+name: spin
+
+on:
+  pull_request:
+    branches: ["main", "v*"]
+    types: ['opened', 'synchronize', 'reopened', 'closed']
+
+jobs:
+  spin:
+    runs-on: ubuntu-latest
+    name: Build and deploy
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Setup `spin`
+        uses: fermyon/actions/spin/setup@v1
+        with:
+          version: canary
+          plugins: js2wasm
+
+      - name: build and deploy
+        uses: fermyon/actions/spin/preview@v1
+        with:
+          manifest_file: example-app/spin.toml
+          fermyon_token: ${{ secrets.FERMYON_CLOUD_TOKEN }}
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          undeploy: ${{ github.event.pull_request && github.event.action == 'closed' }}
```

Comment:
I wonder if we want to explore having this same condition (i.e. PR closed) be the default in the action code?

Reply:
In event section of workflow we need 'closed' type to trigger this behavior. and I think having this in workflow also serves as documenting when this preview will be undeployed

not very strongly against having the default, so if you insist, I will add this as default :)

Reply:
thank you Vaughn